pyproject.toml

```diff
@@ -33,7 +33,7 @@ cwltool = "==3.0.20201109103151"
 dictobj = "*"
 docutils = "*"
 Fabric3 = "*"
-fastapi = "*"
+fastapi = ">=0.63.0"
```

Comment:
I have investigated why poetry was choosing fastapi 0.1.17:
- 0.1.17 was the last version not pinning starlette, see https://github.com/tiangolo/fastapi/commit/3d2c0993c128d406f7f53c1ddbfc9b5d42c08275#
- the latest fastapi (0.63.0) [pins starlette ==0.13.6](https://github.com/tiangolo/fastapi/blob/master/pyproject.toml#L35)
- the latest starlette version is 0.14.2

So I think that poetry chooses starlette 0.14.2 first, and then search for the latest compatible fastapi, which is 0.1.17. So :+1: to this pinning.

Reply:
Yes! Verified:
```
   1: selecting starlette (0.14.2)
   1: fact: fastapi (0.63.0) depends on starlette (0.13.6) ...
   1: derived: not fastapi (==0.63.0)
   1: fact: fastapi (0.62.0) depends on starlette (0.13.6) ...
   1: derived: not fastapi (==0.62.0)
   ...
   1: fact: fastapi (0.1.17) depends on starlette (>=0.9.7) ...   
   1: selecting fastapi (0.1.17)
```
This is not a good approach. I think I would expect it to backtrack and try to downgrade previous nodes, 1 downgrading step at a time. So yes, "-vvv" is very useful! On the bright side, pinning solved it.

Reply:
> This is not a good approach.

The only possible optimal solutions are (starlette 0.13.6 and fastapi 0.63.0) or (starlette 0.14.2 and fastapi 0.1.17), and poetry has no way to know which one is "correct" (i.e. which package to prioritise), given that we depended on both without specifying a minimum version.

Reply:
I think the number of downgrade steps is potentially a good heuristic: downgrade starlette by one minor version or fastapi by 62 minor versions. That said, I've never designed a SAT-solver, so there may be a reason why this heuristic was not used.